backend/jobs/tokenSource.ts

```diff
@@ -0,0 +1,55 @@
+import axios, { AxiosInstance } from 'axios'
+import { apiClientCredentials } from '../api/oauthApi'
+import logger from '../log'
+import errorStatusCode from '../error-status-code'
+
+interface OAuth2Config {
+  clientId: string
+  clientSecret: string
+  url: string
+  timeoutSeconds: number
+  [propName: string]: any
+}
+
+export default class TokenSource {
+  private readonly axiosInstance: AxiosInstance
+
+  private readonly url: string
+
+  constructor(oauth2Config: OAuth2Config) {
+    this.url = oauth2Config.url
+    this.axiosInstance = axios.create({
+      timeout: oauth2Config.timeoutSeconds * 1000,
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        authorization: `Basic ${apiClientCredentials(oauth2Config.clientId, oauth2Config.clientSecret)}`,
```

Comment:
this client doesn't have client credentials - think the tech team tend to like us having separate clients for auth code and client cred grants

Reply:
Not sure what you mean. 2 sets of secrets?  Add another client id and secret especially for this job?

Reply:
that's right - as this requires client credentials

Reply:
Added hmpps-book-video-link-system to auth and switched to using this in code.